contrib/win32/win32compat/misc.c

```diff
@@ -1466,3 +1466,77 @@ is_absolute_path(char *path)
 	
 	return retVal;
 }
+
+/* return 0 - in case of failure */
+BOOL
+create_directory_withsddl(char *path, char *sddl)
+{
+	struct stat st;
+	if (stat(path, &st) < 0) {
+		PSECURITY_DESCRIPTOR pSD = NULL;
+		SECURITY_ATTRIBUTES sa;
+		memset(&sa, 0, sizeof(SECURITY_ATTRIBUTES));
+		sa.nLength = sizeof(SECURITY_ATTRIBUTES);
+		sa.bInheritHandle = FALSE;
+
+		wchar_t *path_w = utf8_to_utf16(path);
+		if (!path_w) {
+			error("%s utf8_to_utf16() has failed to convert string:%s", __func__, path);
```

Comment:
Remember logging is not initialized during service startup. We don't know where this will get logged to.

Reply:
I intentionally added logs so that it can help when we run in debug mode..
When we run as service, logging is not initialized so this wouldn't get logged and it wouldn't create any failures..